index.js

```diff
@@ -3,9 +3,12 @@
 const fp = require('fastify-plugin')
 const { plugin } = require('./lib')
 
-module.exports = fp(function (fastify, options, next) {
-  fastify.addHook('onRequest', plugin(options))
+function fastifyHMAC (fastify, options, next) {
+  fastify.decorateRequest('HMACValidate', plugin(options))
+
```

Comment:
Really my only quibble is the naming of this decorator. Function names _generally_ contain verbs and describe the action that will take place. `HMACValidate` doesn't read to me as an "action".

It looks as though you had this same instinct, as you've named it the way I would have expected it to be named when you decorated the example: [validateHMAC](https://github.com/autotelic/fastify-hmac/pull/17/files#diff-ea03bd13687ba7145ba463540ccb8216R12).

I might use the `validateHMAC` in the plugin then decorate your example with the same name, or another similar action-name like `verifyHMAC`.

Reply:
This is a good point. I've updated the naming accordingly. Thanks!